File: zSDKs/sdk-go/internal/hooks/idempotency.go
package hooks

import (
	"net/http"

	"github.com/google/uuid"
)

type IdempotencyHook struct{}

var _ beforeRequestHook = (*IdempotencyHook)(nil)

func (i *IdempotencyHook) BeforeRequest(hookCtx BeforeRequestContext, req *http.Request) (*http.Request, error) {
	u := uuid.New()

	req.Header.Set("Idempotency-Key", u.String())
	return req, nil
}

File: zSDKs/sdk-typescriptv2/src/hooks/idempotency.ts
import { Awaitable, BeforeRequestContext, BeforeRequestHook } from "./types.js";

import { v4 as uuidv4 } from 'uuid';

export class IdempotencyHook implements BeforeRequestHook {
    beforeRequest(_: BeforeRequestContext, request: Request): Awaitable<Request> {
        request.headers.set("Idempotency-Key", uuidv4());

        return request;
    }
}